Lera_Rodina

valeryrodina opened the pull request.

Commit:
First commit

Commit:
Add new gitignore

Commit:
Add homework1 task

Commit:
Add homework1 task
Renaming the branch

Commit:
Change a structure of project

Commit:
Update after merging

NikolaevArtem commented on gradlew.bat:
```diff
@@ -64,14 +64,29 @@ echo location of your Java installation.
 
 goto fail
 
+:init
```
If they're not needed, please revert changes in gradlew and gradlew.bat,

valeryrodina replied:
thank you! fixed

Commit:
Update after review

Commit:
Add PyramidPrinter task for homework_2

Commit:
Add RandomCharsTable task for homework_2

Commit:
Add RandomCharsTable task for homework_2

Commit:
Refactor PyramidPrinter and RandomCharsTable tasks, add TrafficLight task

Commit:
Update README

Commit:
Fix all 3 tasks

Commit:
Fix for RandomCharsTable task

Commit:
the 2nd Fix for RandomCharsTable task

Commit:
Merge branch 'master' into feature/Lera_Rodina

Commit:
Add link to Coding bat for README.md

Commit:
Add empty ImmutableClass
Change a structure of packages with HWs (add packages with tests for each HW, add Main classes fro all three tasks)
Add new run method for PyramidPrinter task

Commit:
Add requirements for ImmutableClass

Commit:
Add ImmutableClass - RGB (which defines objects that represent colors)

Commit:
Update Immutable class

Commit:
Add new run method for RandomCharsTable task
Update structure for RandomCharsTable

Commit:
Add new run method for TrafficLight task
Update structure for TrafficLight

heptaber commented on src/main/java/homework_2/pyramid_printer/PyramidPrinter.java:
```diff
@@ -0,0 +1,48 @@
+                throw new NumberFormatException();
+            }
+        } catch (NumberFormatException e) {
+            System.out.println("You typed not a number or a negative number, please add a positive number");
```
you should print "Only 1 non-negative integer is allowed as passed parameter" here and above when input value is less then zero.

valeryrodina replied:
thank you! I ve seen updates in Teams in HW description, fixed:)

heptaber commented on src/main/java/homework_2/random_chars_table/RandomCharsTable.java:
```diff
@@ -0,0 +1,84 @@
+        for (int i = 0; i < lengthOfTable; i++) {
+            for (int j = 0; j < widthOfTable; j++) {
+                randomCharsTable[i][j] = (char) ((int) (Math.random() * (maxRand - minRand + 1) + minRand));
+                System.out.print(" | " + randomCharsTable[i][j] + " |");
```
try to avoid printing that extra space in the very start of a row

valeryrodina replied:
fixed, thank you!

heptaber commented on src/main/java/homework_2/random_chars_table/RandomCharsTable.java:
```diff
@@ -0,0 +1,84 @@
+        }
+
+        if ("even".equals(strategy.toLowerCase())) {
+            System.out.print("Even letters: ");
```
```suggestion
            System.out.print("Even letters - ");
```

valeryrodina replied:
fixed. thank you for your attentiveness;)

heptaber commented on src/main/java/homework_2/random_chars_table/RandomCharsTable.java:
```diff
@@ -0,0 +1,84 @@
+        }
+
+        if ("odd".equals(strategy.toLowerCase())) {
+            System.out.print("Odd letters: ");
```
```suggestion
            System.out.print("Odd letters - ");
```

valeryrodina replied:
also fixed!

heptaber commented on src/main/java/homework_2/random_chars_table/RandomCharsTable.java:
```diff
@@ -0,0 +1,84 @@
+            int widthOfTable = inputDataSeparate.nextInt();
+            String strategy = inputDataSeparate.next();
+            if (!"even".equalsIgnoreCase(strategy) && !"odd".equalsIgnoreCase(strategy)) {
+                System.out.println("Please type a correct strategy: even or odd");
```
```suggestion
                System.out.println("Passed parameters should match the format [positive integer] [positive integer] [even|odd]");
```

valeryrodina replied:
yeah, the same wrong text..fixed!

heptaber commented on src/main/java/homework_2/random_chars_table/RandomCharsTable.java:
```diff
@@ -0,0 +1,84 @@
+            }
+            randomCharsTable(lengthOfTable, widthOfTable, strategy);
+        } catch (NegativeArraySizeException | NoSuchElementException e) {
+            System.out.println("You typed negative numbers or strategy/numbers are not added");
```
```suggestion
            System.out.println("Passed parameters should match the format [positive integer] [positive integer] [even|odd]");
```
any exception should print the same message to console

valeryrodina replied:
also fixed:)

heptaber commented on src/main/java/homework_2/random_chars_table/RandomCharsTable.java:
```diff
@@ -0,0 +1,84 @@
+            }
+        }
+
+        stringBuffer.deleteCharAt(stringBuffer.length() - 1);
```
Execution failed for task ':Main.main()'.
what if your 'stringBuffer' is empty? try to run with "1 1 even" or "1 1 odd" as input

valeryrodina replied:
oh thank you! let me check

heptaber commented on src/main/java/homework_2/traffic_light/TrafficLight.java:
```diff
@@ -0,0 +1,58 @@
+    public static final String ANSI_GREEN = "\u001B[32m";
+    public static final String ANSI_YELLOW = "\u001B[33m";
+
+    private static void trafficLight(String inputStr) {
```
you can use it as ExtraTrafficLight.
default TrafficLight realization should provides seconds gone from the day start as input parameter.

valeryrodina replied:
you're right, I thought we could make only one program with hh:mm:ss format.
let me add new app!

heptaber commented on src/main/java/homework_2/traffic_light/TrafficLight.java:
```diff
@@ -0,0 +1,58 @@
+            String str = scanner.nextLine();
+            return str;
+        } catch (NumberFormatException e) {
+            return "Error:(";
```
```suggestion
            return "Only 1 non-negative integer is allowed as passed parameter";
```

valeryrodina replied:
done!

Commit:
Fix text for errors for all 3 tasks
Add tests for PyramidPrinter

Commit:
Move task to new ExtraTrafficLight class
Add TrafficLight task
Fix Main class with choosing mode

Commit:
Add tests for TrafficLight task
Fix bug which tests were found

Commit:
Update README with homework 3 - Immutable class

Commit:
1st iteration: Add tests for RandomCharsTable

NikolaevArtem commented on src/main/java/homework_2/random_chars_table/RandomCharsTable.java:
```diff
@@ -0,0 +1,88 @@
+
+public class RandomCharsTable {
+
+    private static void randomCharsTable(int lengthOfTable, int widthOfTable, String strategy) {
```
![image](https://user-images.githubusercontent.com/58872509/126910486-47e873a1-664f-4d47-b7fc-1f6623aec1d3.png)
There are, it seems, unnecessary whitespaces, please remove

valeryrodina replied:
Thank you! Done:)

NikolaevArtem commented on src/main/java/homework_2/traffic_light/TrafficLight.java:
```diff
@@ -0,0 +1,56 @@
+import java.util.NoSuchElementException;
+import java.util.Scanner;
+
+public class TrafficLight {
```
If i insert 5 7, it doesn't give me an error

valeryrodina replied:
Fixed. Also I added a test for this case

NikolaevArtem commented:
Except small changes, works as expected! Good!

Commit:
Fix RandomCharsTable and TrafficLight tasks after review
Add new test for TrafficLight

Commit:
Fix comment type in ImmutableClass
Add Extramode to main

Commit:
Add CustomFileReader carcass and new structure for hw4

Commit:
Add CustomFileReader task + the method for unit testing

Commit:
Add new package

Commit:
Fix CustomFileReader task,
add tests for CustomFileReader

Commit:
Add Singleton task

Commit:
Finish Singleton task
Add test for it

Commit:
Update README with hw4
Add classes for CustomAnnotation task

Commit:
Add CustomAnnotation task
Add tests for it

Commit:
Add power of number task for hw5 + tests and readme

Commit:
Add CustomRegexMatcher task, update readme, create test directory

Commit:
Add tests for CustomRegexMatcher task

Commit:
Add MapProblemsGenerator task

Commit:
Update README.md
Add empty structure for hw 7 task

Commit:
Add Cat and Kitten classes
Add functional interface

Commit:
Add Main

Commit:
Finish with Main class
Add transformation Kitten to Cat + checking

Commit:
Update README file
Add course project without tests

Commit:
Add About for course project

NikolaevArtem commented on src/main/java/couse_project/About:
```diff
@@ -0,0 +1,30 @@
+
+Course project SEA BATTLE
```
Good description!

NikolaevArtem commented on src/main/java/couse_project/Player.java:
```diff
@@ -0,0 +1,72 @@
+        }
+
+        shoot();
+        return false;
```
What's the case when it will return false? Seems impossible

valeryrodina replied:
let me check and fix if it is needed

NikolaevArtem commented on src/main/java/couse_project/RandomIntelligence.java:
```diff
@@ -0,0 +1,26 @@
+        }
+
+        shoot();
+        return false;
```
Same here: What's the case when it will return false? Seems impossible

valeryrodina replied:
let me check and fix if it is needed

NikolaevArtem commented on src/main/java/couse_project/Game.java:
```diff
@@ -0,0 +1,34 @@
+            displayer.clearScreen();
+            displayer.displayMap(playerOne.getMap());
+            displayer.displayRadar(playerOne.getRadar());
+            if (playerOne.shoot()) {
```
That doesn't look good... usually methods should follow Single Responsibility Prinsiple - it should do only one thing, the thing it promises. In this case it's unexpected it returns boolean. 

Have you tried finishing the game? I'm not sure it will not go to an infinite loop of recursion

valeryrodina replied:
Oh, you're absolutely right. Seems that I forgot to add a condition for loop after last changes for this part:( Let me check

NikolaevArtem commented on src/main/java/couse_project/Game.java:
```diff
@@ -0,0 +1,34 @@
+
+import java.io.IOException;
+
+public class Game {
```
Good! You introduced some abstractions, showed knowledge of Java Core and Java libraries. The code in general looks good, but there are a few cases where I'm not sure about the correct behavior. Also, your project lacks abstract classes/interfaces, and some architecture (the simplest is service/model). The interface is cool, I like it. playing is quite comfortable) I also appreciate automatic ship placement and description, it really helps to test.

Approved!

NikolaevArtem commented on src/main/java/homework_3/ImmutableClass.java:
```diff
@@ -0,0 +1,73 @@
+    Similarly, create copies of your internal mutable objects when necessary to avoid returning the originals in your methods.
+ */
+
+final public class ImmutableClass {
```
The class should have one field, which is link type and mutable. I checked Homework chat right know and it's not clear there, but what I had in mind is this. Please fix

valeryrodina replied:
Fixed!

NikolaevArtem commented on src/main/java/homework_4/custom_annotation/ObjectToJsonConverter.java:
```diff
@@ -0,0 +1,67 @@
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+public class ObjectToJsonConverter {
```
Nice, thorough example, good job!

NikolaevArtem commented on src/main/java/homework_4/custom_file_reader/CustomFileReader.java:
```diff
@@ -0,0 +1,86 @@
+import java.util.List;
+import java.util.Scanner;
+
+public class CustomFileReader {
```
Opt: use try-with-resources, otherwise if you get an exception, stream will not be closed

NikolaevArtem commented on src/main/java/homework_4/custom_file_reader/Main.java:
```diff
@@ -0,0 +1,13 @@
+
+public class Main {
+    public static void main(String[] args) throws IOException {
+        String file = "src/main/resources/custom_file_reader/fileTest.txt";
+        Path path = Paths.get("src/main/resources/custom_file_reader/fileTest.txt");
+        new CustomFileReader(path).run3();
```
Opt: That's quite disturbing, I think Path creation could be done in methods that require it...

NikolaevArtem commented on src/main/java/homework_6/map_problems_generator/Main.java:
```diff
@@ -0,0 +1,22 @@
+
+
+
+        HashMap<MapProblemsMutableGenerator<String>, String> container_mutable_gen = new HashMap<>();
```
There's no mutable key problem in your example
![image](https://user-images.githubusercontent.com/58872509/134233036-96a37b21-0855-4133-8c50-2e515ed13136.png)

valeryrodina replied:
Fixed!

NikolaevArtem commented on src/main/java/homework_7/Main.java:
```diff
@@ -0,0 +1,22 @@
+
+    private static Cat transformToCat(KittenToCatFunction<Cat, Kitten> transformation, Kitten kitten) {
+        return transformation.grow(kitten);
+    }
```
```suggestion
        Cat cat = transform.grow(kitten);
        System.out.println();
        System.out.println(cat);
    }
```

NikolaevArtem commented on src/test/java/homework_4/custom_file_reader/CustomFileReaderTest.java:
```diff
@@ -0,0 +1,169 @@
+public class CustomFileReaderTest extends UnitBase {
+
+    @Test()
+    public void givenWrongFile_whenRun1_thenException() {
```
Nice test cases names!

NikolaevArtem reviewed (changes requested):
Good! The code is generally easy to read, thoroughly tested, has nice examples/description. Minor fixes requested for HW_3 and HW_6.

valeryrodina committed:
Update src/main/java/homework_7/Main.java

Co-authored-by: Artem Nikolaev <Nikolaev.artem.work@yandex.ru>

Commit:
Fix HW3

Commit:
Fix HW3

Commit:
Fix HW6

NikolaevArtem reviewed (approved):
HW approved!
